src/main/java/com/codeborne/selenide/impl/Plugins.java

```diff
@@ -26,8 +26,11 @@
   private static final Map<Class<?>, Object> cache = new ConcurrentHashMap<>();
 
   @SuppressWarnings("unchecked")
-  public static <T> T inject(Class<T> klass) {
-    return (T) cache.computeIfAbsent(klass, Plugins::loadPlugin);
+  public static synchronized <T> T inject(Class<T> klass) {
```

Comment:
looks like not relevant change...

Reply:
@BorisOsipov Yes, it's not related by IE, but it's related to Windows. 
For some reason, this bug happened only on Windows where I was testing my changes on IE browser.

Reply:
to be honest I don't understand why it needs. computeIfAbsent is CAS operations and should guarantee the same semantics...

Reply:
but `loadPlugin` uses `ServiceLoader.load` and it is kind of black magic...

Reply:
That's the point. This is the line from `java.util.concurrent.ConcurrentHashMap#computeIfAbsent` javadoc:
> the computation should be short and simple, and must not attempt to update any other mappings of this map.
> 
> Throws:
> IllegalStateException – if the computation detectably attempts a recursive update to this map that would otherwise never complete

Though, it doesn't fail every time, but occasionally. It seems to depend on order of elements in HashMap or something similar.

Reply:
The problem is not with ServiceLoader.load(). The problem is with our code:
1. inject(Commands.class) calls `new Commands()`
2. which in turn calls `new DownloadFile()`
3. which in turn calls `inject(DownloadFileToFolder.class)`
4. which calls `Plugins.cache.computeIfAbsent()` - during the previous `computeIfAbsent()`. It _sometimes_ causes `IllegalStateException` inside `computeIfAbsent()`.